### ISSUE TITLE
Create album folder with format: %artist% - %album% [%year%]

### ISSUE THREAD

**Issue by syvhmx** [ExtendLord/DeezLoader-Reborn#9]
For example, "Kendrick Lamar - DAMN." can be "Kendrick Lamar - DAMN. (2017)".  
This is to fix the last dot in albums or playlists, unable to delete, even though a fix is made before (?) by removing it automatically. A simple yet effective way if you tend to 'accidentally' break it during any updates.  

Edit: Apparently the dot is still available as of 3.0.1. Hopefully fixed using  in 3.0.2. :+1:

**Comment by ExtendLord** [comment]
I will add a filter for the dot at the end.
I will add %year% as well.

**Comment by syvhmx** [comment]
thanks :+1:

**Comment by ExtendLord** [comment]
#ff8af02

**Comment by ExtendLord** [comment]
ff8af02cb24d573fbe842a49752ac205b251e0ac

**Comment by syvhmx** [comment]
I think I mistakenly tell you.  
The part where the variable %year% is added not to the track names (that's okay), but to the **album folder**  
(Settings only allows editing track names, not folder names)  
(You can't change it in Settings unless you create a new customizable one).  
From this folder name:
```
- Kendrick Lamar - DAMN.
- Shawn Mendes - Illuminate (Deluxe)
- Camilla Cabello - Havana
```

to this folder name:
```
- Kendrick Lamar - DAMN. [2017]
- Shawn Mendes - Illuminate (Deluxe) [2017]
- Camilla Cabello - Havana [2017]
```

Sorry for that :sweat_smile:

**Comment by ExtendLord** [comment]
@syvhmx I will add that too
@syvhmx Great idea

**Comment by syvhmx** [comment]
@ExtendLord Found a fix to do that.
in app.js : (I have to tick 'Create Album Folder in Settings' to enable it)
```
if (settings.createAlbumFolder) {
                            filepath += fixName(settings.createArtistFolder ? metadata.album : `${metadata.artist} - ${metadata.album} [${metadata.year}]`, false) + path.sep;
                            if (!fs.existsSync(filepath)) {
                                fs.mkdirSync(filepath);
```
TLDR; just add [${metadata.year}] to the end of filename.

I don't know how to do it without ticking 'Create Album Folder' in Settings.

**Comment by syvhmx** [comment]
I tried using this:
`downloading.settings.addToPath = downloading.artist + " - " + downloading.name + " [" + downloading.date + "]";`
Output : `Lil Pump - Gucci Gang [undefined]`

**Comment by virtualviolet** [comment]
You do realise that the format %artist% - %album% [%year%] will result in an album spread across multiple folders, some albums have other artists listed as a collaborator on a track or two...

**Comment by syvhmx** [comment]
Well @virtualviolet , when I use my method for example, I try to download one that has tracks with many contributors, the folder name will still show like this:  
`Martin Garrix - So Far Away`
Though the song's artist tag will have like:
`Martin Garrix, David Guetta, Martin Garrix & David Guetta feat. Jamie Scott & Romy Dya`

Just take the first artist name as the base for all the folders.

**Comment by virtualviolet** [comment]
I have found if I disable the artist name and album name folder options in settings.. then it does as you say, takes the first artist name as the base for the entire album.
